keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/config/ClouddriverConfiguration.kt

```diff
@@ -45,13 +45,13 @@ class ClouddriverConfiguration {
   fun clouddriverService(
     clouddriverEndpoint: HttpUrl,
     objectMapper: ObjectMapper,
-    retrofitClient: OkHttpClient
+    clientProvider: OkHttpClientProvider
   ):
     CloudDriverService =
     Retrofit.Builder()
       .addConverterFactory(JacksonConverterFactory.create(objectMapper))
       .baseUrl(clouddriverEndpoint)
-      .client(retrofitClient)
+      .client(clientProvider.getClient(DefaultServiceEndpoint("clouddriver", clouddriverEndpoint.toString())))
```

Comment:
@srekapalli @cfieber This pattern is repeated when instantiating every client. Is this a common enough pattern that it would be worth extracting into an override of `getClient()` that takes just the service name and the endpoint `HttpUrl` object?

Reply:
Yh. Struggled to find a common way to do that because we have several places where we work with Retrofit1 `Endpoint` object as well and some places where we using `String`.   Probably we will revisit this again and settle on a common thing.

Reply:
I see. No worries, thanks for the feedback.